server/api/crud/runtimes/nuclio/function.py

```diff
@@ -293,10 +296,16 @@ def _compile_function_config(
         )
         # since environment variables have a limited size,
         # large serving specs are stored in config maps that are mounted to the pod
+        serving_spec_len = len(serving_spec)
         if (
             can_pass_via_cm
-            and len(serving_spec) >= mlrun.mlconf.httpdb.nuclio.serving_spec_env_cutoff
+            and serving_spec_len >= mlrun.mlconf.httpdb.nuclio.serving_spec_env_cutoff
         ):
+            if serving_spec_len >= SERVING_SPEC_MAX_LENGTH:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"Serving spec length {serving_spec_len} is too large. "
+                    f"The maximum allowed length for the serving spec is {SERVING_SPEC_MAX_LENGTH}."
```

Comment:
I wonder if that exception is useful from a user perspective.
The serving spec is something that is filled behind the scenes when a user calls `serving_function.add_model()`, so will they know that this is a possible cause of the error?